Strange errors with the "template aligment" tutorial

## Your Environment

<!--- Include as many relevant details about the environment you experienced the bug in -->
- Operating System and version: Ubuntu 16.10
- Compiler: gcc version 5.4.0 20160609 (Ubuntu 5.4.0-4ubuntu1) 
- PCL Version: 1.7.2
## Current Behavior

I have used sucesfully the "template aligment" tutorial with the provided data, and the same with some simple point clouds (just a cube), but when going to some real data, I'm getting the next error:

```
[pcl::NormalEstimation::compute] input_ is empty!
[pcl::FPFHEstimation::compute] input_ is empty!
[pcl::FPFHEstimation::initCompute] Init failed.
[pcl::SampleConsensusInitialAlignment::setInputTarget] Invalid or empty point cloud dataset given!
[pcl::SampleConsensusInitialAlignment::setTargetFeatures] Invalid or empty point cloud dataset given!
[pcl::registration::SampleConsensusInitialAlignment::compute] No input target dataset was given!
Violación de segmento (`core' generado)
```

I have get this same error with several point clouds, both as PCD or OBJ files when using them as targets (for the OBJ I changed the call for `pcl::io::loadPCDFile()` to `pcl::io::loadOBJFile()` when testing with them). Point is, when using that same point clouds as templates to be tested (for example using the cube as target) the app doesn't crash, so seems the problem is related to them being used as target.
## Expected Behavior

The tutorial should not crash :-)
## Possible Solution

No idea, but if you could tell me how to enable a debug mode on the library or something, I could be able to take a look on it.
## Code to Reproduce

Use the "template aligment" tutorial with the provided point clouds.
## Context

The idea is to get the transformation matrixes so we can be able later to apply them to the original meshes from co-focal microscope scans (in vrml and/or obj formats) to be able to match them correctly and fix the errors produced during the scanning process.

[templates.txt](https://github.com/PointCloudLibrary/pcl/files/342214/templates.txt)
[cube_template.txt](https://github.com/PointCloudLibrary/pcl/files/342215/cube_template.txt)
[1position1Filament1.obj.txt](https://github.com/PointCloudLibrary/pcl/files/342226/1position1Filament1.obj.txt)
[templates.txt](https://github.com/PointCloudLibrary/pcl/files/342229/templates.txt)
[1position1Filament1 (copia).pcd.txt](https://github.com/PointCloudLibrary/pcl/files/342227/1position1Filament1.copia.pcd.txt)
[7position1Filament1.pcd.txt](https://github.com/PointCloudLibrary/pcl/files/342231/7position1Filament1.pcd.txt)
[1position1Filament1.pcd.txt](https://github.com/PointCloudLibrary/pcl/files/342230/1position1Filament1.pcd.txt)
[1position1Filament1.mtl.txt](https://github.com/PointCloudLibrary/pcl/files/342228/1position1Filament1.mtl.txt)
[cube_template.txt](https://github.com/PointCloudLibrary/pcl/files/342235/cube_template.txt)
[untitled3.pcd.txt](https://github.com/PointCloudLibrary/pcl/files/342233/untitled3.pcd.txt)
[untitled3.mtl.txt](https://github.com/PointCloudLibrary/pcl/files/342236/untitled3.mtl.txt)
[untitled3.obj.txt](https://github.com/PointCloudLibrary/pcl/files/342232/untitled3.obj.txt)
[untitled2.pcd.txt](https://github.com/PointCloudLibrary/pcl/files/342234/untitled2.pcd.txt)
[untitled.pcd.txt](https://github.com/PointCloudLibrary/pcl/files/342237/untitled.pcd.txt)

> Thanks for the bug report and for using the issue template. (joy overflow)

That's what it's maid for, isn't it? ;-)


> So the first thing is to try the same code with pcl HEAD. Since you have 1.7.2 on Ubuntu, I'm assuming you've installed it over the package manager and have all the dependencies already installed, so compiling pcl from source shouldn't be much of a hassle. To enable debug you need to pass the following flag to your cmake command -DCMAKE_BUILD_TYPE=Debug. Let me know if you still have the same issue.

Ok, I have tried several things. First of them, using that flag when compiling the tutorial, without any luck (I still get the same traceback output, instead of one showing me the line where the error was generated or some clue about what's happening).

After that I tried to get HEAD and compile it, there was not problems, but it lasted a long of time on my machine, and worse than that, it rampage memory usage several times and frozen my computer (4GB + 2GB zram) also when compiling with just one core (`make -j1`), needed to reboot it and continue from there. I did this both with the proposed flag and without it (it was not fully clear on your post if I should set the flag when building the library or when building the tutorial), having the same behaviour in both cases (probably don't related, but should be considered...).

After being able to compile HEAD with debug support enabled, I installed it globally (`sudo make install`), setting it at `/usr/local`. After that I used `cmake` to regenerate the tutorial, getting the next error:

```
CMake Warning at CMakeLists.txt:11 (add_executable):
  Cannot generate a safe linker search path for target template_alignment
  because files in some directories may conflict with libraries in implicit
  directories:

    link library [libpcl_common.so] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /usr/local/lib
    link library [libpcl_octree.so] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /usr/local/lib
    link library [libpcl_io.so] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /usr/local/lib
    link library [libpcl_kdtree.so] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /usr/local/lib
    link library [libpcl_search.so] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /usr/local/lib
    link library [libpcl_sample_consensus.so] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /usr/local/lib
    link library [libpcl_filters.so] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /usr/local/lib
    link library [libpcl_features.so] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /usr/local/lib
    link library [libpcl_segmentation.so] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /usr/local/lib
    link library [libpcl_visualization.so] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /usr/local/lib
    link library [libpcl_surface.so] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /usr/local/lib
    link library [libpcl_registration.so] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /usr/local/lib
    link library [libpcl_keypoints.so] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /usr/local/lib
    link library [libpcl_tracking.so] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /usr/local/lib
    link library [libpcl_recognition.so] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /usr/local/lib
    link library [libpcl_outofcore.so] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /usr/local/lib
    link library [libpcl_people.so] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /usr/local/lib

  Some of these libraries may not be found correctly.
```

Seems cmake got a conflict when trying to find the PointCloud library between the one from the Ubuntu package and the one I've compiled, the point is that also by removing the ones from Ubuntu the error persist. Anyway, the `Makefile` is generated, so I exec `make` to build the tutorial and got the next error:

```
[ 50%] Building CXX object CMakeFiles/template_alignment.dir/template_alignment.cpp.o
/media/piranna/Elements/HBP/PointClouds/template_alignment.cpp: In member function ‘void TemplateAlignment::align(FeatureCloud&, TemplateAlignment::Result&)’:
/media/piranna/Elements/HBP/PointClouds/template_alignment.cpp:164:61: warning: ‘void pcl::Registration<PointSource, PointTarget, Scalar>::setInputCloud(const PointCloudSourceConstPtr&) [with PointSource = pcl::PointXYZ; PointTarget = pcl::PointXYZ; Scalar = float; pcl::Registration<PointSource, PointTarget, Scalar>::PointCloudSourceConstPtr = boost::shared_ptr<const pcl::PointCloud<pcl::PointXYZ> >]’ is deprecated: [pcl::registration::Registration::setInputCloud] setInputCloud is deprecated. Please use setInputSource instead. [-Wdeprecated-declarations]
       sac_ia_.setInputCloud (template_cloud.getPointCloud ());
                                                             ^
In file included from /usr/local/include/pcl-1.8/pcl/registration/registration.h:607:0,
                 from /usr/local/include/pcl-1.8/pcl/registration/ia_ransac.h:43,
                 from /media/piranna/Elements/HBP/PointClouds/template_alignment.cpp:14:
/usr/local/include/pcl-1.8/pcl/registration/impl/registration.hpp:43:1: note: declared here
 pcl::Registration<PointSource, PointTarget, Scalar>::setInputCloud (
 ^
make[2]: *** No hay ninguna regla para construir el objetivo '/usr/lib/x86_64-linux-gnu/libpcl_common.so', necesario para 'template_alignment'.  Alto.
CMakeFiles/Makefile2:67: fallo en las instrucciones para el objetivo 'CMakeFiles/template_alignment.dir/all'
make[1]: *** [CMakeFiles/template_alignment.dir/all] Error 2
Makefile:83: fallo en las instrucciones para el objetivo 'all'
make: *** [all] Error 2
```

For some reason it's trying to link against the one from the Ubuntu package... that don't exist anymore :-/ Do you have any clue about what could be happening here?

--- a/doc/tutorials/content/template_alignment.rst
+++ b/doc/tutorials/content/template_alignment.rst
@@ -133,6 +133,10 @@ Next we load the target cloud (from the filename supplied on the command line).
 
 We then perform a little pre-processing on the data to get it ready for alignment.  The first step is to filter out any background points.  In this example we assume the person we're trying to align to will be less than 1 meter away, so we apply a pass-through filter, filtering on the "z" field (i.e., depth) with limits of 0 to 1.
 
+.. note:: This is application and data dependent. You may need to tune the
+          threshold (or drop this filter entirely) to make it work with your
+          data.
+
 .. literalinclude:: sources/template_alignment/template_alignment.cpp
    :language: cpp
    :lines: 253-260

